### PR TITLE
Remove potential NULLCHECK creation after LIR on arm32.

### PR DESCRIPTION
**Pull request by sandreenko**
@erozenfeld noticed that changes from https://github.com/dotnet/runtime/pull/39824 could work wrong on arm32 because we could create a NullCheck node after Lower phase. If it reaches codegen it will fail there. I don't have a repro test, this path is probably unreachable for arm32 currently, but it is better to use a newly added function from #40226 to handle it.

Also, use `TYP_INT` for the new `NULLCHECK` nodes, before in some places we were using `TYP_BYTE`. The actual type doesn't matter, codegen ignores it.

No libs diffs.

### CODE REVIEW

**Comment by sandreenko** [comment]
PTAL @erozenfeld @CarolEidt @dotnet/jit-contrib

File: src/coreclr/src/jit/compiler.cpp [hunk 1]
```diff
@@ -9293,11 +9293,16 @@ bool Compiler::lvaIsOSRLocal(unsigned varNum)
 //    tree       - the node to change;
 //    basicBlock - basic block of the node.
 //
+// Notes:
+//    the function should not be called after lowering for platforms that does not support
```
Comment (erozenfeld) [inline src/coreclr/src/jit/compiler.cpp:9297]:
```suggestion
//    the function should not be called after lowering for platforms that do not support

**Review by erozenfeld — APPROVED** [review]
LGTM

**Review by CarolEidt — APPROVED** [review]
LGTM, though I believe that historically `GT_NULLCHECK`s have always been `TYP_BYTE` (not sure why, since codegen ignores it) so I'm just curious why you changed it.

**Comment by sandreenko** [comment]
> LGTM, though I believe that historically GT_NULLCHECKs have always been TYP_BYTE (not sure why, since codegen ignores it) so I'm just curious why you changed it.

We are changing it to `TYP_INT` in `TransformUnusedIndirection` (and before that we were changing it to `TYP_INT` in `LowerIndir`, https://github.com/dotnet/runtime/pull/40226/files). So it was a little confusing that we were changing it once to `TYP_INT`, then calling `gtChangeOperToNullCheck` and changing it to `TYP_BYTE`. 

I  could make a change in `TransformUnusedIndirection` and use `TYP_BYTE` there instead.

**Comment by CarolEidt** [comment]
> I could make a change in TransformUnusedIndirection and use TYP_BYTE there instead.

No need; I was just curious about the reasoning. Thanks

**Pull request merged**